### PR TITLE
New source: mixednovel.net

### PR DESCRIPTION
**Pull request by hoangphuc05**
Issue: #1623

### CODE REVIEW

File: sources/en/m/mixednovel.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+from bs4 import Tag
+
+from lncrawl.core.crawler import Crawler
+
+logger = logging.getLogger(__name__)
+search_url = "%s/?s=%s&post_type=wp-manga"
+
+
+class ListNovelCrawler(Crawler):
```
Comment (dipu-bd) [inline sources/en/m/mixednovel.py:13]: please change the class name to `MixedNovelNet`

File: sources/en/m/mixednovel.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    base_url = [
+        "https://mixednovel.net/",
+    ]
+
+    def initialize(self):
+        self.init_executor(1)
+
+    def search_novel(self, query):
+        query = quote_plus(query.lower())
+        soup = self.get_soup(search_url % (self.home_url.strip("/"), query))
```
Comment (dipu-bd) [inline sources/en/m/mixednovel.py:24]: f-string is preferred: `f"{self.home_url}?s={query}&post_type=wp-manga"`

File: sources/en/m/mixednovel.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        results = []
+        for tab in soup.select(".c-tabs-item .c-tabs-item__content"):
+            a = tab.select_one(".post-title a")
+            alternative = tab.select_one(".mg_alternative .summary-content").text.strip()
+            author = tab.select_one(".mg_author .summary-content a").text.strip()
+            results.append(
+                {
+                    "title": a.text,
+                    "url": self.absolute_url(a["href"]),
+                    "info": "Alternative title(s): %s | Author(s): %s" % (alternative, author),
```
Comment (dipu-bd) [inline sources/en/m/mixednovel.py:34]: f-string is preferred: `f"Alternative: {alternative} | Author(s): {author}"`

File: sources/en/m/mixednovel.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        logger.info("%s", self.novel_author)
+
+        possible_image = soup.select_one(".summary_image img")
+        if isinstance(possible_image, Tag):
+            urls = possible_image['src']
+            self.novel_cover = self.absolute_url(urls) if urls else None
+        logger.info("Novel cover: %s", self.novel_cover)
+
+        chapter_url = "%s/ajax/chapters/"
+        chapter_soup = self.post_soup(chapter_url % (self.novel_url.strip("/")))
```
Comment (dipu-bd) [inline sources/en/m/mixednovel.py:60]: use f-string: `f"{self.novel_url}ajax/chapters/"`

File: sources/en/m/mixednovel.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                    "id": chap_id,
+                    "title": a.text.strip(),
+                    "url": self.absolute_url(a["href"]),
+                }
+            )
+
+    def download_chapter_body(self, chapter):
+        soup = self.get_soup(chapter["url"])
+        contents = soup.select(".reading-content .text-left")
+        return "".join([str(p) for p in contents])
```
Comment (dipu-bd) [inline sources/en/m/mixednovel.py:76]: question: why not use `self.cleaner.extract_contents(contents)`?

**Comment by hoangphuc05** [comment]
thanks for the feedback, code updated

**Pull request merged**